src/Agent.Worker/Release/Artifacts/BuildArtifact.cs

```diff
@@ -123,6 +125,8 @@ public IArtifactDetails GetArtifactDetails(IExecutionContext context, AgentArtif
                     AccessToken = accessToken,
                     Project = projectId.ToString(),
                     TfsUrl = new Uri(tfsUrl),
+                    DefinitionName = artifactDetails["DefinitionName"],
```

Comment:
check for existence of key before usage? if agent is new & service is old, may be the key wouldn't exist. in which case it is fine to not check for this existence.